aloha-core/src/main/scala/com/eharmony/aloha/semantics/compiled/plugin/csv/CsvModelRunner.scala

```diff
@@ -528,21 +691,35 @@ object CsvModelRunner {
             case StringType  => instantiate[String]
         }
 
-        (s: String) => m(fn(s))
+        ((s: String) => fn(s), m)
     }
 
-    def getRunnableInfo(config: Option[CsvModelRunnerConfig]) = config map { case conf =>
-        val inputType = conf.inputType.right.get.get
-
-        val fn = predictionFunction(inputType, conf.outputType, conf.imports, conf.classCacheDir, conf.model.get)
-
-        val out = conf.outputFile.map(f => VFS.getManager.resolveFile(f).getContent.getOutputStream).getOrElse(System.out)
-        val closeOut = conf.outputFile.isDefined
-
-        val is = conf.inputFile.map(f => VFS.getManager.resolveFile(f).getContent.getInputStream).getOrElse(System.in)
-        val in = scala.io.Source.fromInputStream(is).getLines()
-
-        (conf, in, fn, out, closeOut)
+    def getPredictionOutputFormat(config: Option[CsvModelRunnerConfig]): Option[PredictionOutputFormat] = config map { case conf =>
+        val inputType: InputType = conf.inputType.right.get.get
+
+        // (String) => Nothing
+        // Model[Nothing, Any]
+        val (inF, model) = inputAndModel(inputType, conf.outputType, conf.imports, conf.classCacheDir, conf.model.get)
+
+        val out: OutputStream = conf.outputFile.map(f => VFS.getManager.resolveFile(f).getContent.getOutputStream).getOrElse(System.out)
+        val closeOut: Boolean = conf.outputFile.isDefined
+
+        val is: InputStream = conf.inputFile.map(f => VFS.getManager.resolveFile(f).getContent.getInputStream).getOrElse(System.in)
+        val in: Iterator[String] = scala.io.Source.fromInputStream(is).getLines()
```

Comment:
Yeah it is sucker!  https://github.com/scala/scala/blob/v2.10.5/src/library/scala/io/Source.scala#L96-Lundefined